ci_framework/plugins/action/ci_script.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+
+# Copyright Red Hat, Inc.
+# Apache License Version 2.0 (see LICENSE
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+action: ci_script
+
+short_description: Runs a string given script remotly while providing logging
+
+description:
+- Discover latest image matching a specific pattern on remote server.
+
+options:
```

Comment:
Since we're consuming `ansible.builtin.script` module under the hood, I'd rather support the module parameter as listed here:
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/script_module.html

Some of them are really nice, such as the `creates` and `removes`.
Maybe the `chdir` is better "masked", since it should be overridden within this action plugin anyway.